File: backend/BeElite/src/main/java/com/salesianos/triana/BeElite/dto/User/UserDto.java
package com.salesianos.triana.BeElite.dto.User;

import com.salesianos.triana.BeElite.model.Usuario;
import jakarta.transaction.Transactional;
import jakarta.validation.constraints.NotEmpty;
import lombok.Builder;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Builder
public record UserDto(
        @NotEmpty(message = "Athlete must have username.")
        String username,
        @NotEmpty(message = "Athlete must have a name.")
        String name,
        String profilePicUrl,
        @NotEmpty(message = "Athlete must have email.")
        String email,
        LocalDate joinedProgramDate
) {

    @Transactional
    public static UserDto of(Usuario u){
        return UserDto.builder()
                .username(u.getUsername())
                .name(u.getName())
                .email(u.getEmail())
                .profilePicUrl(u.getProfilePicUrl())
                .joinedProgramDate(u.getJoinDate().toLocalDate())
                .build();
    }

    public static UserDto of(Usuario u, LocalDate joinedProgramDate){
        return UserDto.builder()
                .username(u.getUsername())
                .name(u.getName())
                .email(u.getEmail())
                .profilePicUrl(u.getProfilePicUrl())
                .joinedProgramDate(joinedProgramDate)
                .build();
    }

    public static UserDto empty(){
        return UserDto.builder().build();
    }
}

File: frontend/be_elite/lib/models/Coach/user_dto.dart
class UserDto {
  String? username;
  String? name;
  String? profilePicUrl;
  String? email;
  String? joinedProgramDate;

  UserDto({this.username, this.name, this.profilePicUrl, this.email, this.joinedProgramDate});

  factory UserDto.fromJson(Map<String, dynamic> json) => UserDto(
        username: json['username'] as String?,
        name: json['name'] as String?,
        profilePicUrl: json['profilePicUrl'] as String?,
        email: json['email'] as String?,
        joinedProgramDate: json['joinedProgramDate'] as String?,
      );

  Map<String, dynamic> toJson() => {
        'username': username,
        'name': name,
        'profilePicUrl': profilePicUrl,
        'email': email,
        'joinedProgramDate': joinedProgramDate,
      };
}
